### ISSUE TITLE
[PHP7.4] ReferencedNameHelper::createAllReferencedNames produces warnings on php multiline comment in HEREDOC

### ISSUE THREAD

**Issue by enl** [slevomat/coding-standard#1355]
**Steps to reproduce:**

1. Create a example.php file with the following code:

```php
<?php

<<<EXAMPLE
/**
 * I'm a multiline comment inside HEREDOC
 * Why do I even exist?
 * One can create a sniff that checks copyright correctness and want to provide an example :)
 */
EXAMPLE;
```

2. Run `phpcs --standard=SlevomatCodingStandard example.php`

**Expected Results:**

No warnings.

**Actual results:**

```
PHP Warning:  Unterminated comment starting line 1 in /Users/aleksandrpanshin/work/modules/phpcs-rulesets/vendor/slevomat/coding-standard/SlevomatCodingStandard/Helpers/ReferencedNameHelper.php on line 470

Warning: Unterminated comment starting line 1 in /Users/aleksandrpanshin/work/modules/phpcs-rulesets/vendor/slevomat/coding-standard/SlevomatCodingStandard/Helpers/ReferencedNameHelper.php on line 470
```

**Notes**

According to my investigation, it happens because multiline heredoc creates several tokens and they are searched for references one by one in ReferencedNameHelper::createAllReferencedNames in this piece:

```php
// Find referenced names inside double quotes string
if (self::isNeedParsedContent($tokens[$nameStartPointer]['code'])) {
$content = $tokens[$nameStartPointer]['content'];

if (self::isNeedParsedContent($tokens[$nameStartPointer - 1]['code'])) {
	$content = '"' . $content;
}
if (self::isNeedParsedContent($tokens[$nameStartPointer + 1]['code'])) {
	$content .= '"';
}
$names = self::getReferencedNamesFromString($content);
foreach ($names as $name) {
	$referencedNames[] = new ReferencedName($name, $nameStartPointer, $nameStartPointer, ReferencedName::TYPE_CLASS);
}
$beginSearchAtPointer = $nameStartPointer + 1;
continue;
}
```

Which in process calls `self::getReferencedNamesFromString("/**");`, and this function calls `token_get_all("/**");`, which produces the warning because comment is indeed not closed.

The issue started once I updated to 7.1.

**Comment by kukulich** [comment]
I'm not able to reproduce the bug. Can you please try to write failing test?

**Comment by enl** [comment]
Yup, checked build status:

The warning is there for 7.4 but neither for 8.0 nor 8.1.

**Comment by kukulich** [comment]
Ok, so that's probably bug in PHPCS - the tokens should be same on all PHP versions.

**Comment by enl** [comment]
Yeah, sounds logical. I'll create an issue on their repo. 

Although, question is what is *expected* list of tokens: one we get on 7.4 or one we get on 8.+ :)
